### ISSUE TITLE
Add RustSec DB source

### ISSUE THREAD

**Issue by wagoodman** [anchore/grype#529]
Add the ability to match rust crates against the RustSec DB https://github.com/rustsec/advisory-db

**Comment by westonsteimel** [comment]
@wagoodman, there is also https://github.com/google/osv which aggregates all of these sources into a [common schema](https://github.com/ossf/osv-schema)

**Comment by westonsteimel** [comment]
So how do we actually go about adding new data sources for grype?  Is the existing process documented somewhere?

**Comment by westonsteimel** [comment]
Ah, so would it be somewhere in https://github.com/anchore/anchore-engine/tree/master/anchore_engine/services/policy_engine/engine/feeds?

**Comment by Shnatsel** [comment]
Hello, RustSec advisory DB maintainer here :wave: 

RustSec uses a custom TOML-based format, but we also provide all advisories in the [OSV format](https://github.com/ossf/osv-schema) in real time. So if you support OSV already, supporting RustSec should be very easy. You can get the advisory data from Google's API or [straight from our git repo](https://github.com/rustsec/advisory-db/tree/osv).

RustSec uses [SemVer precedence rules](https://github.com/rustsec/advisory-db) so matching should be quite straightforward.

Please let me know if you have any questions, run into any issues, etc.

**Comment by 100tomer** [comment]
+1 I’m also very interested to hear if there’s been any progress. For me, scanning Rust crates with the current Grype version consistently returns zero findings, even when I can manually locate multiple vulnerabilities in RustSec.

**Comment by Shnatsel** [comment]
That is surprising - RustSec exports their data to [OSV](https://osv.dev/), so anything using OSV API should be able to access RustSec data.

@100tomer does running [cargo audit](https://crates.io/crates/cargo-audit) surface any vulnerabilities?

**Comment by 100tomer** [comment]
> That is surprising - RustSec exports their data to [OSV](https://osv.dev/), so anything using OSV API should be able to access RustSec data.
> 
> [@100tomer](https://github.com/100tomer) does running [cargo audit](https://crates.io/crates/cargo-audit) surface any vulnerabilities?

Thanks for the quick reply. That was my mistake, I now see that both cargo audit and Grype do detect the vulnerability.
I think the issue on my side is that Grype seems to require a Cargo.lock file (I am guessing).
Is there a way to scan Cargo crates without a Cargo.lock, for example by providing the crate file or the package name and version directly?

Thanks

Edit: quick update, generating an SBOM manually with the package metadata works great. I’m still curious though whether Grype can scan crates or just a Cargo.toml alone without requiring a lockfile

**Comment by Shnatsel** [comment]
No, scanning based on Cargo.toml alone is not possible even in theory, because it specifies only possible version ranges rather than a specific version for any given package.

**Comment by willmurphyscode** [comment]
Grype supports Rust through GHSA today. For example:

```
$ grype db search --vuln GHSA-rjr4-v43m-pxq6
VULNERABILITY        PACKAGE    ECOSYSTEM   NAMESPACE             VERSION CONSTRAINT
GHSA-rjr4-v43m-pxq6  triton-vm  rust-crate  github:language:rust  >=0.41.0,<2.0.0
```

At first site https://osv.dev/list?q=&ecosystem=crates.io seems to have ~2x as many things as https://github.com/advisories?query=type%3Areviewed+ecosystem%3Arust but I'm concerned there will be duplicate data, for example https://osv.dev/vulnerability/RUSTSEC-2026-0004 and https://github.com/advisories/GHSA-rjr4-v43m-pxq6 appear to be the same vuln, but the GHSA has a RustSet alias, while the RustSec/OSV listing does not have a GHSA alias.

I think this is worth pursuing still if:

1. We can prove that the OSV RustSec data is fresher or more complete than GHSA (I suspect it is, but the first two RustSec entries in osv.dev I checked both had GHSAs already reviewed.)
2. We can devise a way to deduplicate from GHSA. (Right now the aliases only point one way, where GHSA provides a RustSec ID, but RustSec doesn't (always?) provide a GHSA ID.)

But I also think Rust is covered reasonably well by GHSA data in Grype today.
